lesson02.md

```diff
@@ -0,0 +1,440 @@
+<!-- .slide: id="lesson2" -->
+
+# Basic Frontend - Fall 2024
+
+Lesson 2, Tuesday, 2024-09-17
+
+---
+
+### Recap
+
+What did we talk about last lesson?
+
+---
+
+### Recap: Javascript Console
+
+`F12` is the magic key
+
+---
+
+### Recap: data types
+
+Number:
+```js
+42
+-3.1415
+```
+
+
+We can also do computation
+<!-- .element: class="fragment" -->
+
+```js
+1 + 1
+
+10 / 5
+
+5 ** 3
+```
+<!-- .element: class="fragment" -->
+
+
+---
+
+### Recap: data types
+
+String:
+```js
+"Alan"
+'Berlin'
+```
+
+---
+
+### String operators
+
+The addition operator (`+`) can be used to combine (concatenate) two strings:
+
+```js
+"a" + "b"    // results: "ab"
+```
+
+---
+
+### String operators
+
+The addition operator can be used on Strings and Numbers:
+
+```js
+1 + 1                // is 2
+"hello" + " world"   // is "hello world"
+```
+
+---
+
+### Importance of data types
+
+```js
+1 + 1        // is 2
+"1" + "1"    // is "11"
+```
+
+It is important to select the appropriate data type for your data, as the behavior of operators in JavaScript varies depending on the data type used.
+
+---
+
+### Small exercise
+
+use the `+` operator to combine multiple strings to generate the following string. Use your name and favorite food:
+
+`Hello, my name is + (name) + and I like + (food)`
+
+---
+
+### String operators
+
+We shouldn't use operators (like `-`, `/`, `*`) on strings. While JavaScript allows that, the results are unexpected. With strings, use `+` for concatenation.
+
+---
+
+### Data Type: Boolean
+
+* **Boolean** describes a yes/no situation
+* in JavaScript, we use `true` or `false`
+
+---
+
+### Practice
+
+Open the console, and type in the following:
+
+* `true`
+* `false`
+
+Why would we dedicate an entire data type for just two values?
+<!-- .element: class="fragment" -->
+
+---
+
+### Comparison operators
+
+* `===` strict equality
+
+```js
+1 === 1     // true
+3 === 10    // false
+
+3 === "Hello"  // false
+3 === "3"      // false
+
+"Ada" === "Ada"   // true
+"Ada" === "Alan"  // false
+"Alan" === "alan" // false
+```
+
+---
+
+### Comparison operators
+
+* `!==` strict inequality
+
+```js
+3 !== 5           // true
+3 !== 3           // false
+1 !== "Cookies"   // true
+```
+
+---
+
+### Comparison operators
+
+We can use `===` and `!==` to test for the equality and inequality of any 2 values in javascript.
```

Comment:
ultra-nit: but one could aim for consistency and always write JavaScript. Its written like that in many places, but not everywhere. What do you think?

Reply:
Nice catch. Fixed.